src/ristretto/ristretto_keys.rs

```diff
@@ -219,6 +219,13 @@ impl RistrettoPublicKey {
             compressed: pk.compress(),
         }
     }
+
+    pub(crate) fn new_from_compressed(compressed: CompressedRistretto) -> Option<RistrettoPublicKey> {
+        let c = compressed.clone();
+        compressed
+            .decompress()
+            .map(|point| RistrettoPublicKey { compressed: c, point })
+    }
```

Comment:
```suggestion
        compressed
            .decompress()
            .map(|point| RistrettoPublicKey { compressed, point })
    }
```

Reply:
Would this even compile? Will try

Reply:
Applied. I thought it was a move, but it's a borrow